src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metrics/AllMetrics.java

```diff
@@ -129,6 +130,44 @@ public String toString() {
     }
   }
 
+  public enum CacheCustomDimension implements MetricDimension {
+    CACHE_TYPE(Constants.TYPE_VALUE);
+
+    private final String value;
+
+    CacheCustomDimension(String value) {
+      this.value = value;
+    }
+
+    @Override
+    public String toString() {
+      return value;
+    }
+
+    public static class Constants {
+      public static final String TYPE_VALUE = "CacheType";
+    }
+  }
+
+  public enum CacheCustomValue implements MetricValue {
+    CACHE_MAX_SIZE(Constants.MAX_SIZE);
+
+    private final String value;
+
+    CacheCustomValue(String value) {
+      this.value = value;
+    }
+
+    @Override
+    public String toString() {
+      return value;
+    }
+
+    public static class Constants {
+      public static final String MAX_SIZE = "Cache_MaxSize";
```

Comment:
should we follow the  naming convention  and rename this to CACHE_MAX_SIZE_VALUE ?

Reply:
Agree, updating this to `CACHE_MAX_SIZE_VALUE`